src/base.jl

```diff
@@ -657,6 +692,26 @@ function get_components_by_name(
     return components
 end
 
+"""
+    get_components_forecasts(
+                            ::Type{T},
+                            sys::System,
+                            initial_time::Dates.DateTime,
+                            ) where T <: Component
+
+Get the forecasts of a component of type T with initial_time.
```

Comment:
Questions:
1. Should the docstring call out the fact that you will get back forecasts of all types?
2. Should there be an arg or kwarg that allows the user to specify a specific forecast type?  You could default it to Forecast.

Reply:
I am working on this, I prefer to include the error if T is abstract. I think it's fine to return forescasts of different types.